GooglePay/src/main/java/com/braintreepayments/api/googlepay/GooglePayRequest.kt

```diff
@@ -0,0 +1,316 @@
+ * `false` otherwise.
+ *
+ * Optional.
+ * @property shippingAddressRequirements the shipping address requirements. [ShippingAddressRequirements]
+ *
+ * Optional.
+ * @property allowPrepaidCards `true` prepaid cards are allowed, `false` otherwise.
+ *
+ * Defines if PayPal should be an available payment method in Google Pay.
+ * Defaults to `true`.
+ * @property enablePayPal `true` by default. Allows PayPal to be a payment method in Google Pay.
+ *
+ * Optional.
+ * @property merchantName The merchant name that will be presented in Google Pay
+ *
+ * ISO 3166-1 alpha-2 country code where the transaction is processed. This is required for
+ * merchants based in European Economic Area (EEA) countries.
+ * NOTE: to support Elo cards, country code must be set to "BR"
+ * @property countryCode The country code where the transaction is processed
+ *
+ * Optional
+ * @property totalPriceLabel Custom label for the total price within the display items
+ */
+@Suppress("TooManyFunctions")
+@Parcelize
+class GooglePayRequest(
+    // NEXT_MAJOR_VERSION: allow merchants to set transaction info params individually and build
+    // JSON object under the hood
+    var transactionInfo: TransactionInfo? = null,
+    var isEmailRequired: Boolean = false,
+    var isPhoneNumberRequired: Boolean = false,
+    var isBillingAddressRequired: Boolean = false,
+    @BillingAddressFormat
+    var billingAddressFormat: Int = 0,
+    var isShippingAddressRequired: Boolean = false,
+    // NEXT_MAJOR_VERSION: allow merchants to set shipping address requirements params individually
+    // and build JSON object under the hood
+    var shippingAddressRequirements: ShippingAddressRequirements? = null,
+    var allowPrepaidCards: Boolean = false,
+    var isPayPalEnabled: Boolean = true,
+    var googleMerchantName: String? = null,
+    var countryCode: String? = null,
+    var totalPriceLabel: String? = null,
+    var allowCreditCards: Boolean = true,
+    private var environment: String? = null,
+    private val allowedPaymentMethods: MutableMap<String, String> = HashMap(),
+    private val tokenizationSpecifications: MutableMap<String, String> = HashMap(),
+    private val allowedAuthMethods: MutableMap<String, String> = HashMap(),
+    private val allowedCardNetworks: MutableMap<String, String> = HashMap()
+) : Parcelable {
+
+    fun setEnvironment(environment: String?) {
+        this.environment =
+            if ("PRODUCTION" == environment?.uppercase(Locale.getDefault())) "PRODUCTION" else "TEST"
+    }
+
+    fun getEnvironment(): String? {
+        return environment
+    }
+
+    /**
+     * Simple wrapper to assign given parameters to specified paymentMethod
+     *
+     * @param paymentMethodType The paymentMethod to add to
+     * @param parameters        Parameters to assign to the paymentMethod
+     */
+    fun setAllowedPaymentMethod(paymentMethodType: String, parameters: JSONObject) {
```

Comment:
Do these setter functions match the original Java interface? Would this be a breaking change for merchants?

Reply:
The function signature does not change for this function.